--- a/src/backports-3.14.22-1/backport-include/linux/compat.h
+++ b/src/backports-3.14.22-1/backport-include/linux/compat.h
@@ -14,8 +14,8 @@
 #endif
 
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(3,4,0))
-#define compat_put_timespec LINUX_BACKPORT(compat_put_timespec)
-extern int compat_put_timespec(const struct timespec *, void __user *);
+//#define compat_put_timespec LINUX_BACKPORT(compat_put_timespec)
+//extern int compat_put_timespec(const struct timespec *, void __user *);
 #endif
 
 #endif /* __BACKPORT_COMPAT_H */
